Assets/VRTK/Prefabs/SnapDropZone/VRTK_SnapDropZone.cs

```diff
@@ -316,6 +316,68 @@ public virtual VRTK_InteractableObject GetCurrentSnappedInteractableObject()
             return currentSnappedObject;
         }
 
+        /// <summary>
+        /// The Clone method returns the GameObject of the cloned snap drop zone
+        /// </summary>
+        /// <param name="position">Position of the cloned GameObject</param>
+        /// <returns>The GameObject of the clone</returns>
+        public virtual GameObject Clone(Vector3 position)
+        {
+            VRTK_SnapDropZone cloneSDZ = Instantiate(gameObject, position, transform.rotation).GetComponent<VRTK_SnapDropZone>();
+
+            for (int childID = 0; childID < cloneSDZ.transform.childCount; childID++)
+            {
+                Transform child = cloneSDZ.transform.GetChild(childID);
+                if (child.GetComponent<VRTK_InteractableObject>() != null)
+                    Destroy(child.gameObject);
+            }
+
+            if (isSnapped)
+            {
+                VRTK_InteractableObject currObject = currentSnappedObject;
+
+                //Get copy of Objects original state
+                Transform previousParent;
+                bool previousKinematic;
+                bool previousGrabbable;
+                currObject.GetPreviousState(out previousParent, out previousKinematic, out previousGrabbable);
+
+                GameObject clonedObject = null;
+                if (cloneNewOnUnsnap)
+                {
+                    clonedObject = Instantiate(objectToClone);
+                    clonedObject.SetActive(true);
+                }
+                else
+                {
+                    clonedObject = Instantiate(currObject.gameObject);
+                }
+
+                clonedObject.transform.position = cloneSDZ.transform.position;
+                cloneSDZ.ForceSnap(clonedObject);
+
+                StartCoroutine(OverridePreviousStateAtEndOfFrame(clonedObject.GetComponent<VRTK_InteractableObject>(), previousParent, previousKinematic, previousGrabbable));
+            }
+
+            return cloneSDZ.gameObject;
+        }
+
+        private IEnumerator OverridePreviousStateAtEndOfFrame(VRTK_InteractableObject io, Transform parent, bool kinematic, bool grabbable)
```

Comment:
shouldn't be `private`  should be `protected virtual` and should be grouped with the other `protected virtual` methods